marine/marine.py

```diff
@@ -233,6 +330,47 @@ def _parse_output(output: POINTER(c_char_p), length: int) -> List[Optional[str]]
             for i in range(length)
         )
 
+    @classmethod
+    def _load_marine_packet(cls, field):
+        if field.children:
+            return cls._load_child_fields(field.children.contents)
+        else:
+            return cls._load_field_value(field.value)
+
+    @classmethod
+    def _load_child_fields(cls, children):
+        child_fields = {}
+        for child in children.data[: children.len]:
+            name = child.name.decode("utf-8")
+            value = cls._load_marine_packet(child)
+            child_fields[name] = value
+        return child_fields
+
+    @classmethod
+    def _load_field_value(cls, value):
+        value_len = value.len
+        value_type = value.type
+        if value_type == 0 or value_len <= 0:
+            return None
+
+        if value_type == MarinePacketFieldValueType.INT:
+            return value.int_value
+
+        if value_type == MarinePacketFieldValueType.UINT:
+            return value.uint_value
+
+        if value_type == MarinePacketFieldValueType.BOOL:
+            return bool(value.bool_value)
+
+        if value_type == MarinePacketFieldValueType.STR:
+            return value.str_value.decode("utf-8")
+
+        if value_type == MarinePacketFieldValueType.BYTES:
+            return value.str_value[:value_len]
+
+        if value_type == MarinePacketFieldValueType.LIST:
+            return [cls._load_field_value(v) for v in value.list_value[:value_len]]
+
```

Comment:
Do we want an `else` here to throw an exception or something?

Reply:
Agreed